test/object-store/sync/flx_sync.cpp

```diff
@@ -793,6 +793,19 @@ TEST_CASE("flx: client reset", "[sync][flx][app][client reset]") {
             })
             ->run();
     }
+
+    SECTION("DiscardLocal: completion callbacks fire after client reset") {
```

Comment:
Was this test something you discovered during this work? How is it related?

Reply:
indeed it is. `wait_for_download` hangs without the fix.